go-controller/pkg/node/gateway_iptables.go

```diff
@@ -79,6 +79,18 @@ func deleteIptRules(rules []nodeipt.Rule) error {
 	return nodeipt.DelRules(rules)
 }
 
+// ensureChain ensures that a chain exists within a table
+func ensureChain(table, chain string) error {
+	for _, proto := range clusterIPTablesProtocols() {
+		ipt, err := util.GetIPTablesHelper(proto)
+		if err != nil {
+			return fmt.Errorf("failed to get IPTables helper to add UDN chain: %v", err)
+		}
+		addChaintoTable(ipt, table, chain)
```

Comment:
ah! I was talking about addChainToTable...OK looks like there isn't a wrapper for that and hence this happens..

Reply:
I know it is unrelated to this change but `addChaintoTable` ignores all errors, not only the ones where the chain already exists. I feel like it would be best to introduce a new method to the helper, something like:
```
// EnsureChain creates a new chain if it doesn't already exist
func (ipt *IPTables) EnsureChain(table, chain string) error {
	exists, err := ipt.ChainExists(table, chain)
	if err != nil || exists {
		return err
	}
	return ipt.NewChain(table, chain)
}
```
This doesn't have to happen in this PR.